stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/GiphyMediaAttachmentView.kt

```diff
@@ -96,27 +99,66 @@ public class GiphyMediaAttachmentView : ConstraintLayout {
             it.imagePreviewUrl ?: it.titleLink ?: it.ogUrl
         } ?: return
 
-        val height = if (style.giphyType == GiphyInfoType.ORIGINAL) {
-            giphyInfo?.height ?: GIPHY_INFO_DEFAULT_HEIGHT_DP.dpToPx()
+        if (style.sizingMode == GiphySizingMode.ADAPTIVE) {
+            applyAdaptiveSizing(
+                giphyInfo = giphyInfo,
+                giphyUrl = url
+            )
         } else {
-            (giphyInfo?.height ?: GIPHY_INFO_DEFAULT_HEIGHT_DP).dpToPx()
+            applyFixedSizing(
+                giphyUrl = url
+            )
         }
-        val width = giphyInfo?.width
+    }
 
-        this.updateLayoutParams {
-            this.height = height
-            if (width != null && style.giphyType == GiphyInfoType.ORIGINAL) {
-                this.width = width
-            }
+    /**
+     * Automatically resizes the Giphy while respecting the Giphy's
+     * aspect ratio.
+     *
+     * This method is meant to ignore [GiphyMediaAttachmentViewStyle.width],
+     * [GiphyMediaAttachmentViewStyle.height] and [GiphyMediaAttachmentViewStyle.dimensionRatio].
+     *
+     * @param giphyInfo Used to determine the aspect ratio of the Giphy.
+     * @param giphyUrl The URL to the Giphy we are loading.
+     */
+    private fun applyAdaptiveSizing(giphyInfo: GiphyInfo?, giphyUrl: String) {
+        binding.root.updateConstraints {
+            constrainMaxWidth(binding.imageView.id, ViewGroup.LayoutParams.MATCH_PARENT)
+            val ratio = (giphyInfo?.width ?: 1).toFloat() / (giphyInfo?.height ?: 1).toFloat()
+            setDimensionRatio(binding.imageView.id, ratio.toString())
+
+            constrainViewToParentBySide(binding.imageView, ConstraintSet.LEFT)
+            constrainViewToParentBySide(binding.imageView, ConstraintSet.RIGHT)
+            constrainViewToParentBySide(binding.imageView, ConstraintSet.TOP)
+            constrainViewToParentBySide(binding.imageView, ConstraintSet.BOTTOM)
         }
-        binding.imageView.updateLayoutParams {
-            this.height = height
-            if (width != null && style.giphyType == GiphyInfoType.ORIGINAL) {
-                this.width = width
+
+        loadGiphy(giphyUrl)
+    }
+
+    /**
+     * Applies fixed sizing to the Giphy container set via
+     * [GiphyMediaAttachmentViewStyle].
+     *
+     * @param giphyUrl The URL to the Giphy we are loading.
+     */
+    private fun applyFixedSizing(giphyUrl: String) {
+        binding.root.updateConstraints {
+            constrainWidth(binding.imageView.id, style.width)
+
+            when {
+                // If the user has set a height, respect the height
```

Comment:
Slightly improved after the PR was approved.

Please do not automatically resolve this comment & merge but rather retest the fixed sized testing section and then merge.

Reply:
Tested with:

```
        <item name="streamUiGiphyMediaAttachmentSizingMode">fixed_size</item>
        <item name="streamUiGiphyMediaAttachmentWidth">200dp</item>
        <item name="streamUiGiphyMediaAttachmentHeight">150dp</item>
```

Also tested with either set height, no width, set width, no height, no width or height, default settings. 

All look good and scroll really fast and clean!